testing/sdk_metrics.py

```diff
@@ -70,11 +83,61 @@ def get_metrics(package_name, service_name, task_name):
     raise Exception("No metrics found")
 
 
-def wait_for_any_metrics(package_name, service_name, task_name, timeout):
-    def metrics_exist():
-        log.info("verifying metrics exist for {}".format(service_name))
-        service_metrics = get_metrics(package_name, service_name, task_name)
-        # there are 2 generic metrics that are always emitted
-        return len(service_metrics) > 2
+def extract_metric_names(service_name, service_metrics):
```

Comment:
Rather than putting per-service cases here, how about having the per-service tests define a callback function that they pass in to `wait_for_service_metrics`? Maybe a system like:
- `wait_for_service_metrics`'s `expected_metrics` list parameter could be replaced with a list of callbacks (one per expected metric).
- `wait_for_service_metrics` would then pass the list of names it gets to each of the callbacks, waiting for each of them to return true _at least once_.
- The callbacks themselves would internally do whatever service-specific mangling is needed on a given metric name

Or alternately a simpler solution could be:
- The `expected_metrics` parameter could be a list of regexes rather than plain strings